src/main/java/com/scalar/db/storage/multistorage/MultiStorageConfig.java

```diff
@@ -0,0 +1,116 @@
+package com.scalar.db.storage.multistorage;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+import com.scalar.db.config.DatabaseConfig;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Properties;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public class MultiStorageConfig {
+
+  public static final String PREFIX = DatabaseConfig.PREFIX + "multi_storage.";
+  public static final String STORAGES = PREFIX + "storages";
+  public static final String TABLE_MAPPING = PREFIX + "table_mapping";
+  public static final String DEFAULT_STORAGE = PREFIX + "default_storage";
+
+  private final Properties props;
+
+  private Map<String, DatabaseConfig> databaseConfigMap;
+  private Map<String, String> tableStorageMap;
+  private String defaultStorage;
+
+  public MultiStorageConfig(File propertiesFile) throws IOException {
+    this(new FileInputStream(propertiesFile));
+  }
+
+  public MultiStorageConfig(InputStream stream) throws IOException {
+    props = new Properties();
+    props.load(stream);
+    load();
+  }
+
+  public MultiStorageConfig(Properties properties) {
+    props = new Properties(properties);
+    load();
+  }
+
+  public Properties getProperties() {
+    return props;
+  }
+
+  private void load() {
+    String storage = props.getProperty(DatabaseConfig.STORAGE);
+    if (storage == null || !storage.equals("multi-storage")) {
```

Comment:
I think it would be probably better to make "multi-storage" a constant since it's used in several places.